aspnetcore/signalr/authn-and-authz.md

```diff
@@ -103,6 +95,25 @@ var connection = new HubConnectionBuilder()
 
 Windows Authentication is only supported by the browser client when using Microsoft Internet Explorer or Microsoft Edge.
 
+### Using Claims to Customize Identity Handling
+
+An app that authenticates users can derive SignalR user IDs from user claims. To specify how SignalR creates user IDs, implement `IUserIdProvider` and register the implementation.
+
+The sample code demonstrates how you would use claims to select the user's email address as the identifying property. 
+
+[!code-csharp[Email provider](authn-and-authz/sample/EmailBasedUserIdProvider.cs?name=EmailBasedUserIdProvider)]
+
+In the sample code, the account registration code contains the following line of code, which adds a claim with type `ClaimsTypes.Email` to the ASP.NET identity database. 
+
+[!code-csharp[Adding the email to the ASP.NET identity claims](authn-and-authz/sample/pages/account/Register.cshtml.cs?name=AddEmailClaim)]
+
+The sample code contains both the name and email implementations. Register this component in your `Startup.ConfigureServices` method **after** the call to `.AddSignalR`. 
```

Comment:
Doesn't need to be after AddSignalR anymore I believe.

Reply:
@BrennanConroy Was that a change in 2.2?

Reply:
Nope, in a patch in 2.1